packages/core/src/user_session_store.ts

```diff
@@ -36,10 +36,10 @@ export class UserSessionStore {
     this.sessionTokenExpirationInterval =
       userSessionOptions.session.lifetime ||
       DEFAULT_SESSION_EXPIRATION_INTERVAL;
-    this.sampling = userSessionOptions.sampling === undefined ?  DEFAULT_SAMPLING : userSessionOptions.sampling;
+    this.sampling = userSessionOptions.sampling === undefined ? DEFAULT_SAMPLING : userSessionOptions.sampling;
 
-    if (!getCookie('EA_SESSION_SAMPLED')) {
-      this.updateSessionSampledExpire();
+    if (!getCookie('EA_SESSION_SAMPLED') || (!this.isSessionSampled() && this.sampling === 1)) {
```

Comment:
I suggest to improve code readability by instead updating `updateSessionSampledExpire`, do:

- check whether the current session is not sampled and current sampling rate is 1
- if true, set the cookie sample to true and update expiry
- if false, continue previously to the original `updateSessionSampledExplre`